services/paymaster/src/api.rs

```diff
@@ -223,11 +223,10 @@ fn get_domain_name(
 }
 
 fn get_domain_state<'a>(
-    state: &'a ServerState,
+    domains_guard: &'a HashMap<String, DomainState>,
```

Comment:
you don't need the `_guard` naming here since the type in the function is just a reference to a hashmap